### PR TITLE
BREAKING(collections): deprecate `groupBy()`

### PR DESCRIPTION
**Pull request by NotWoods**
`groupBy` can be replaced by `Object.groupBy` in Deno 1.37. Creating this as a draft for now because the TypeScript lib isn't updated, but running this without typechecking passes fine.

### CODE REVIEW

**Comment by lino-levan** [comment]
Related to #3594.

File: collections/group_by.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // This module is browser compatible.
 
 /**
+ * @deprecated Use `Object.groupBy` instead.
```
Comment (iuioiua) [inline collections/group_by.ts:5]:
```suggestion
 * @deprecated (will be removed in 0.207.0) Use {@linkcode Object.groupBy} instead.
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]
This change looks reasonable and good to me. Thanks!

**Comment by kt3k** [comment]
Deprecation makes sense to me, but safari still doesn't support this according to MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy

I'm afraid we might get similar reaction to the case of `readableStreamFromIterable()` https://github.com/denoland/deno_std/pull/3579#issuecomment-1700637852

**Comment by lino-levan** [comment]
Not sure about this. I'm still considering `std/shim` or `std/ponyfill` or something.

**Comment by kt3k** [comment]
We updated the deprecation policy in https://github.com/denoland/deno_std/pull/3807

Now we recommend the users to pin the URL to the old std version if they still need them.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**